internal/controller/isbservicerollout_controller.go

```diff
@@ -89,6 +93,8 @@ func (r *ISBServiceRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Re
 			return ctrl.Result{}, err
 		}
 	}
+	// generate metrics for isbServiceRollout
+	r.customMetrics.IncISBServiceMetrics(isbServiceRollout.Name, isbServiceRollout.Namespace)
```

Comment:
Should we do this here or below after we successfully reconciled? If we can't reconcile this, then this number effectively represents the number of ISBServiceRollouts rather than ISBServices I guess? That makes me wonder which one we actually want here. In the future, we will have this idea of a Rollout which can temporarily have 2 different children as a mechanism for gradually replacing one with the other. Perhaps if we just name this with "Rollout" or "RO" in the name, then we're good here?

Reply:
Yes, I agree I also think that it should go down after "successful reconciled" as this metrics is for ISB Service count not for Rollout